### ISSUE TITLE
[Phase 5] Training Pipeline

### ISSUE THREAD

**Issue by Sakeeb91** [Sakeeb91/speech-enhancement-system#6]
## Phase 5: Training Pipeline

**Parent:** #1
**Depends on:** #4, #5

### Objectives
Implement complete training pipeline with validation, checkpointing, and logging.

### Tasks

- [ ] Create training loop with progress bar
- [ ] Implement validation loop
- [ ] Add learning rate scheduling (ReduceLROnPlateau)
- [ ] Add early stopping
- [ ] Implement gradient clipping
- [ ] Add model checkpointing
- [ ] Log training metrics
- [ ] Support resuming from checkpoint

### Files to Create/Modify

| File | Action | Description |
|------|--------|-------------|
| `src/train.py` | Create | Training script |
| `configs/train_config.yaml` | Create | Training configuration |

### Training Configuration

```python
config = {
    'model': 'spectral',  # or 'unet'
    'batch_size': 16,
    'learning_rate': 0.0001,
    'epochs': 100,
    'patience': 10,
    'lr_patience': 5,
    'lr_factor': 0.5,
    'grad_clip': 5.0,
    'segment_length': 32000,
    'num_workers': 4,
}
```

### Code Reference

```python
import torch
from torch.optim.lr_scheduler import ReduceLROnPlateau
from tqdm import tqdm

def train_epoch(model, loader, optimizer, loss_fn, device, grad_clip):
    model.train()
    total_loss = 0

    for noisy, clean in tqdm(loader, desc='Training'):
        noisy = noisy.to(device)
        clean = clean.to(device)

        optimizer.zero_grad()
        enhanced = model(noisy)
        loss = loss_fn(enhanced, clean)
        loss.backward()

        torch.nn.utils.clip_grad_norm_(model.parameters(), grad_clip)
        optimizer.step()

        total_loss += loss.item()

    return total_loss / len(loader)


def validate(model, loader, loss_fn, device):
    model.eval()
    total_loss = 0

    with torch.no_grad():
        for noisy, clean in loader:
            noisy = noisy.to(device)
            clean = clean.to(device)
            enhanced = model(noisy)
            loss = loss_fn(enhanced, clean)
            total_loss += loss.item()

    return total_loss / len(loader)
```

### Definition of Done

- [ ] Training loop runs without errors
- [ ] Validation loss computed each epoch
- [ ] LR scheduler reduces LR on plateau
- [ ] Early stopping triggers when validation loss stagnates
- [ ] Best model checkpoint saved
- [ ] Training can resume from checkpoint
- [ ] Progress bar shows loss values

### Technical Notes

**For junior developers:**
- Use `model.train()` and `model.eval()` to set correct mode
- `torch.no_grad()` saves memory during validation
- Gradient clipping prevents exploding gradients in RNNs
- Save optimizer state too for proper resume
